linked_list/linked_list.py

```diff
@@ -23,13 +23,46 @@ def __init__(self):
     self.tail = None
 
   def add_to_tail(self, value):
-    pass
+    node = Node(value)
+    if self.tail is not None:
+      self.tail.set_next(node)
+    else:
+      self.head = node
+    self.tail = node
 
   def remove_head(self):
-    pass
+    if self.head is not None:
+      val = self.head.value
+      newHead = self.head.next_node
+      if newHead is None:
+        self.tail = None
+      else: 
+        self.tail = newHead.next_node
+      self.head = newHead
+      return val
+    else: 
+      return None
+      # We never set tail to something that isn't none
 
+      
   def contains(self, value):
-    pass
+    if self.head is not None:
+      curr_node = self.head
+      while True:
```

Comment:
make sure you watch your number of while loops as those can severely decrement the time complexity of your methods. Code is clean, and also great job on your methods as well!